lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2616,8 +2616,35 @@ size_t TypeSystemSwiftTypeRef::GetIndexOfChildMemberWithName(
 
 size_t
 TypeSystemSwiftTypeRef::GetNumTemplateArguments(opaque_compiler_type_t type) {
-  return m_swift_ast_context->GetNumTemplateArguments(ReconstructType(type));
+  auto impl = [&]() -> size_t {
+    using namespace swift::Demangle;
+    Demangler dem;
+    NodePointer node = DemangleCanonicalType(dem, type);
+
+    if (!node)
+      return 0;
+
+    switch (node->getKind()) {
+    case Node::Kind::BoundGenericClass:
+    case Node::Kind::BoundGenericEnum:
+    case Node::Kind::BoundGenericStructure:
+    case Node::Kind::BoundGenericProtocol:
+    case Node::Kind::BoundGenericOtherNominalType:
+    case Node::Kind::BoundGenericTypeAlias:
+    case Node::Kind::BoundGenericFunction: {
+      auto child = node->getChild(1);
```

Comment:
I am pretty sure `getChild` fails when accessing an index that doesn't exist, so generally we defensively check the size before accessing it.

also llvm style is to indicate pointers even with `auto` declarations, ex `auto *child`:

Reply:
One question I had about children: can I assume the indexes of the children won't change between different compiler releases? For example, right now `TypeList` is always the child at index 1 of these `BoundGenericX` nodes, can I assume this will always be true, or should I iterate over the children until I find the first one with kind `TypeList`?

Reply:
I believe this is ABI and so shouldn't change. I think we assume indexes in some places, and in other places we iterate. I think assuming a known index is fine, but let's see what Adrian says.

Reply:
```
      if (node->getNumChildren() > 1) {
        NodePointer child =  getChild(1);
```

Reply:
The ordering is ABI, but there are some node types that by design some nodes can have an arbitrary number of children and future extensions might add additional children, I believe.